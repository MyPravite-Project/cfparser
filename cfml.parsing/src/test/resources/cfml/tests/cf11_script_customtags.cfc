   // cfm file based custom tags
      Cf_happybirthday(name="john", birthdate="December 5, 1987");
