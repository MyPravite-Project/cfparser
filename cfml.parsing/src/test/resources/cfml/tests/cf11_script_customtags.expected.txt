/*===TOKENS===*/
LINE_COMMENT             <// cfm file based custom tags
>
IDENTIFIER               <Cf_happybirthday>
'('                      <(>
IDENTIFIER               <name>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <john>
CLOSE_STRING             <">
','                      <,>
IDENTIFIER               <birthdate>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <December 5, 1987>
CLOSE_STRING             <">
')'                      <)>
';'                      <;>
/*===TREE===*/
(scriptBlock
  (element
    (statement
      (startExpression
        (compareExpression
          (baseExpression
            (unaryExpression
              (memberExpression
                (functionCall
                  (identifierOrReservedWord (identifier Cf_happybirthday))
                  (
                  (argumentList
                    (argument
                      (identifier name)
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (primaryExpression (literalExpression (stringLiteral " john ")))
                            )
                          )
                        )
                      )
                    )
                    ,
                    (argument
                      (identifier birthdate)
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (primaryExpression (literalExpression (stringLiteral " December 5, 1987 ")))
                            )
                          )
                        )
                      )
                    )
                  )
                  )
                )
              )
            )
          )
        )
      )
      ;
    )
  )
  (endOfScriptBlock <EOF>)
)
/*======*/